### DISCUSSION TITLE
helm controller ready for production?

### DISCUSSION THREAD

**Discussion by avielb** [fluxcd/flux2#4071] [General] [answered]
hey, 
found this section in the docs:
https://fluxcd.io/flux/migration/helm-operator-migration/#is-the-helm-controller-ready-for-production

do you think it is still relevant?
i am currently in the work of planning migration from flux v1 to flux v2 for all of our environments and that includes our main production environment which currently relies heavily on helm operator. 

and I wanted to personally thank everyone involved in the project ❤️ 
the docs as well as the product are written amazing and very useful.

**Comment by kingdonb** [comment, marked as answer]
> Probably

This word and the notes after it was written in March 2021, in this guide at a time when Flux v2 was still newly considered "production ready" and it's worth asking the question, is any of this still relevant, because the entire guide to migrating from Flux v1 was written so long ago!

Thanks for the kind words about the Flux docs. 🦖🚀

If you are still using Helm Operator, I'm sure or at least hope that you're already aware of the support status as it has been archived for some time, and can no longer be supported. I recommend upgrading ASAP since Helm Operator runs on old Kubernetes and Helm versions that cannot be patched anymore, and are probably loaded with CVEs.

> we do not guarantee there will not be any breaking changes before we reach General Availability.
> We are however committed to provide [conversion webhooks] for upcoming API versions.

> There may be (internal) behavioral changes in upcoming releases, but they should be aimed at further stabilizing the Helm Controller itself

These statements are all still relevant and I think they're remarkably accurate for having been drafted as they are written over a year ago, and worth noting that in the [roadmap](https://fluxcd.io/roadmap) Helm is the focus of the upcoming Flux 2.1 release.

Stay tuned for upcoming changes in Helm Controller, that will culminate in the final stabilization of the v2 HelmRelease API.

The next minor Flux release is anticipated to bring more updates to the Helm Controller, and you can expect some changes to land between now and then. "Nothing earth-shattering" could be the headline, if you've been keeping up with Flux news and following our roadmap, I think what's being delivered is quite what was promised. There will be a CRD change to a stable API version, and a few feature updates marking GA some recent bits that may have been highly anticipated or were teased in earlier releases of Helm Controller, like Drift Correction that is already available in Flux 2.0, behind a feature flag that enables upgrading based on Drift Detection triggers that monitor Helm-created resources for changes since they were installed.

Now there is certainly more clarity about what will be in 2.1 than there was at the time of this writing, and I'm sure I'm not covering it all, read the issues queue and the release notes when they come out for a fuller picture. There were more details explained in the latest Flux dev meeting, that you can also catch on youtube.com/@fluxcd if you want to hear the details right from the horse's mouth.

I'd say the fact that it says "Probably" production ready is definitely a statement from some time ago, while now Helm Controller definitely has many production users. We have had some discussion about how to better hide these docs about migrating from Flux v1, so that people who are upgrading from Flux2 v0.x to Flux 2.x.x don't find it. It is good that you checked in, so we are aware of some users still on Helm Operator and Flux v1! Please let us know if there's anything we can do to help smooth your transition. 🥀🔥🤠
